docs/source/tests/pytest.rst

```diff
@@ -0,0 +1,124 @@
+
+.. code-block:: python
+
+    ''' tests.test_core.test_xy '''
+    import pytest
+    import numpy as np
+
+    from floodlight.core.xy import XY
+
+    # creation of the fixture
+    @pytest.fixture()
+    def example_xy_data_pos_int() -> np.ndarray:
+        positions = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])
+        return positions
+
+    # testing a function with the fixture being passed as an argument
+    def test_x_pos_int(example_xy_data_pos_int: np.ndarray) -> None:
+        # Arrange
+        data = XY(example_xy_data_pos_int)
+
+        # Act
+        x_position = data.x
+
+        # Assert
+        assert np.array_equal(x_position, np.array([[1, 3], [5, 7]]))
+
+Fixtures are a quite powerful tool since they are modular and can also request other fixtures.
+
+When to create fixtures?
+~~~~~~~~~~~~~~~~~~~~~~~~
+In case you are writing multiple tests that all make use of the same underlying test data, then it can be advantageous to create a fixture. Otherwise it is common to arrange the data inside your testing function.
+
+Where to create fixtures?
+~~~~~~~~~~~~~~~~~~~~~~~~~
+With the pytest framework there are different possibilities where the fixtures can be implemented. Creating fixtures in different locations only serves to clarify the test environment, especially when working collaboratively in a team. The following options are common solutions:
+
+    #. Inside the testing files.
+    #. Inside a ``conftest.py`` file.
+    #. Inside an extra file which is then integrated into the ``conftest.py`` file as a plugin.
+
+The ``conftest.py`` file just follows a naming convention of pytest and enables to share fixtures across multiple files. The fixtures implemented inside the ``conftest.py`` file can be accessed from testing files laying in the same folder layer or in a subdirectory without any import. For more detailed information (especially on option 3.) have a look on this `link <https://docs.pytest.org/en/6.2.x/fixture.html>`_.
+
+Marks
+------
+Marks can be used to categorize your tests. To do so you need to decorate the method with ``@pytest.mark.<mark_name>``. When executing the ``pytest -m <mark_name>`` command (:doc:`see here </tests/testing>`) only methods decorated with ``@pytest.mark.<mark_name>`` will be selected for the testing. This can be advantageous if you have tests that are slower because they are for example accessing a database but you want to quickly run your test suite.
```

Comment:
The `see here` link will break if joined into one file

Reply:
all :doc: links where changed to :refs: